server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -195,6 +203,22 @@ public CoordinatorDynamicConfig(
         "maxNonPrimaryReplicantsToLoad must be greater than or equal to 0."
     );
     this.maxNonPrimaryReplicantsToLoad = maxNonPrimaryReplicantsToLoad;
+
+    if (maxSegmentsToLoad == null) {
```

Comment:
https://github.com/apache/druid/pull/11135#discussion_r620474349 See this discussion behind reasoning for this handling